web/src/lib/components/forms/edit-user-form.svelte

```diff
@@ -70,6 +71,22 @@
       isShowResetPasswordConfirmation = false;
     }
   };
+
+  function generatePassword(length: number = 16) {
+    let generatedPassword = '';
+
+    const characterSet = '0123456789' + 'abcdefghijklmnopqrstuvwxyz' + 'ABCDEFGHIJKLMNOPQRSTUVWXYZ' + ',.-{}+!#$%/()=?';
+
+    for (let i = 0; i < length; i++) {
+      let randomNumber = crypto.getRandomValues(new Uint32Array(1))[0];
+      randomNumber = randomNumber / 2 ** 32;
+      randomNumber = Math.floor(randomNumber * characterSet.length);
+
+      generatedPassword += characterSet[randomNumber];
+    }
+
+    return generatedPassword;
```

Comment:
A bit lengthy for my liking, what do you think of this?

```suggestion
    const characterSet = '0123456789' + 'abcdefghijklmnopqrstuvwxyz' + 'ABCDEFGHIJKLMNOPQRSTUVWXYZ' + ',.-{}+!#$%/()=?';

    return Array.from(crypto.getRandomValues(new Uint32Array(length)))
      .map((x) => characterSet[x % characterSet.length])
      .join('');
````

Reply:
Both of these generators are biased because of using modulo / floor. The lower `N % len(characterSet)` characters will be chosen more often than others because the wrap around is most likely not aligned.

To make a secure one, you should either use a uniform random number distribution to pick characters or just make a loop and generate random bytes while discarding any that are not a valid characters. Looping is probably the easiest to get right, cryptographically speaking.

Reply:
The server just uses random bytes and then stips out non alpha-numeric characters, which seems more than sufficient imo.

Reply:
Generating stuff like this client-side has its complexities. I added a note that we should move this to the server in a follow-up PR.